src/libcharon/plugins/kernel_super/system_events/system_events.c

```diff
@@ -0,0 +1,53 @@
+
+#include "system_events.h"
+
+#include <stdlib.h>
+#include <utils/debug.h>
+
+static system_events_obj_t *g_system_events = NULL;
```

Comment:
There's no reason to have a global object that holds the system_events, they can be held as a data member of ` struct kernel_super_ipsec_t`.
In particular:
1. Remove this file
2. Rename 'system_events_interface.h' to 'system_events.h' (replace this file) - It should contain the API for system_events